modules/lsp/default.nix

```diff
@@ -2,13 +2,76 @@
   lib,
   config,
   options,
+  pkgs,
   ...
 }:
 let
   inherit (lib) types;
   inherit (lib.nixvim) toLuaObject;
 
   cfg = config.lsp;
+
+  mkServerSubmodule =
+    {
+      name ? null,
+      package ? null,
+    }:
+    types.submodule [
+      ./server.nix
+      {
+        options.package =
+          let
+            extraDescription = ''
+
+
+              Alternatively, the language server should be available on your `$PATH`.
+            '';
+          in
+          if package == null then
+            lib.mkOption {
+              type = types.nullOr types.package;
+              default = null;
+              description = ''
+                Package to use for this language server.
+                ${extraDescription}
+              '';
+            }
+          else if lib.isOption package then
+            package
+          else
+            lib.mkPackageOption pkgs name {
+              nullable = true;
+              default = package;
+              inherit extraDescription;
+            };
+      }
+    ];
```

Comment:
I think we can move this logic back into `server.nix` if we use [`importApply`](https://noogle.dev/f/lib/modules/importApply) to pass in additional package-related args